File: java-to-kotlin/src/main/java/nl/alimeshkat/recipes/data/IngredientsEntity.java
package nl.alimeshkat.recipes.data;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.jetbrains.annotations.NotNull;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Table(name = "ingredients")
public final class IngredientsEntity {
    @ManyToOne(optional = false)
    @JoinColumn(name = "recipes_id", nullable = false)
    private RecipesEntity recipes;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long ingredientId;

    @NotNull
    private String name;
    @NotNull
    private String type;
    @NotNull
    private Integer weight;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        IngredientsEntity that = (IngredientsEntity) o;

        if (getIngredientId() == null || that.getIngredientId() == null) return false;

        return getIngredientId().equals(that.getIngredientId());
    }

    @Override
    public int hashCode() {
        if (getIngredientId() == null) {
            return super.hashCode();
        } else {
            return getIngredientId().hashCode();
        }
    }
}

File: java-to-kotlin-complete/src/main/kotlin/nl/alimeshkat/recipes/data/IngredientsEntity.kt
package nl.alimeshkat.recipes.data

import jakarta.persistence.*

@Entity
@Table(name = "ingredients")
class IngredientsEntity(
    @ManyToOne(optional = false)
    @JoinColumn(name = "recipes_id", nullable = false)
    val recipes: RecipesEntity? = null,
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val ingredientId: Long? = null,
    val name: String,
    val type: String,
    val weight: Int,
) {

    override fun equals(other: Any?): Boolean {
        return when {
            other !is IngredientsEntity -> false
            this === other -> true
            ingredientId == null || other.ingredientId == null -> false
            else -> ingredientId == other.ingredientId
        }
    }

    override fun hashCode(): Int {
        return ingredientId?.hashCode() ?: super.hashCode()
    }
}
